### ISSUE TITLE
py-jupyterlab-server not working properly when installed using spack

### ISSUE THREAD

**Issue by mdorier** [LLNL/spack#20173]
When I install the `py-jupyterlab-server` package with spack, then run the jupyterlab server and create or open a Python notebook, jupyter fails to find launch a Python kernel. When instead I install the `py-pip` package with spack, then load it and install `jupyterlab_server` using pip, I don't have this problem. 

### Steps to reproduce the issue

```console
$ spack install py-jupyterlab-server
$ spack load -r py-jupyterlab-server
$ jupyter notebook --ip 0.0.0.0 --port 8888 --no-browser
```
Following this command, open a browser at the address `http://localhost:8888`, and create a new Python notebook. This will open a new page with the notebook, but the python kernel will fail to start. The log on the server shows the following:

```
[I 14:33:53.684 NotebookApp] Kernel started: f44e6e16-aac3-47e9-84c0-2940365f499d, name: python3
/projects/spack/opt/spack/linux-debian10-sandybridge/gcc-8.3.0/python-3.7.4-7jkb4dss6icmzzors2yx7egepyr77ght/bin/python3.7: No module named ipykernel_launcher
[I 14:33:56.679 NotebookApp] KernelRestarter: restarting kernel (1/5), new random ports
/projects/spack/opt/spack/linux-debian10-sandybridge/gcc-8.3.0/python-3.7.4-7jkb4dss6icmzzors2yx7egepyr77ght/bin/python3.7: No module named ipykernel_launcher
[I 14:33:59.686 NotebookApp] KernelRestarter: restarting kernel (2/5), new random ports
/projects/spack/opt/spack/linux-debian10-sandybridge/gcc-8.3.0/python-3.7.4-7jkb4dss6icmzzors2yx7egepyr77ght/bin/python3.7: No module named ipykernel_launcher
[I 14:34:02.691 NotebookApp] KernelRestarter: restarting kernel (3/5), new random ports
/projects/spack/opt/spack/linux-debian10-sandybridge/gcc-8.3.0/python-3.7.4-7jkb4dss6icmzzors2yx7egepyr77ght/bin/python3.7: No module named ipykernel_launcher
[I 14:34:05.698 NotebookApp] KernelRestarter: restarting kernel (4/5), new random ports
/projects/spack/opt/spack/linux-debian10-sandybridge/gcc-8.3.0/python-3.7.4-7jkb4dss6icmzzors2yx7egepyr77ght/bin/python3.7: No module named ipykernel_launcher
[W 14:34:08.703 NotebookApp] KernelRestarter: restart failed
[W 14:34:08.703 NotebookApp] Kernel f44e6e16-aac3-47e9-84c0-2940365f499d died, removing from map.
[W 14:34:14.739 NotebookApp] Replacing stale connection: f44e6e16-aac3-47e9-84c0-2940365f499d:e4524f5aacdb43b190c48a1687e19de6
[W 14:34:36.727 NotebookApp] Replacing stale connection: f44e6e16-aac3-47e9-84c0-2940365f499d:e4524f5aacdb43b190c48a1687e19de6
[W 14:34:53.727 NotebookApp] Timeout waiting for kernel_info reply from f44e6e16-aac3-47e9-84c0-2940365f499d
[E 14:34:53.728 NotebookApp] Error opening stream: HTTP 404: Not Found (Kernel does not exist: f44e6e16-aac3-47e9-84c0-2940365f499d)
```
Yet I can see that the `py-ipykernel` package is installed and loaded.

The following installation of jupyterlab-server with pip works fine:

```console
$ spack install py-pip
$ spack load -r py-pip
$ pip install --user jupyterlab_server
$ jupyter notebook --ip 0.0.0.0 --port 8888 --no-browser
```

### Information on your system

I'm on a Debian machine, and my `packages.yaml` file forces spack to install Python 3.7.4.

**Comment by adamjstewart** [comment]
I've had this problem before. Make sure all of the dependencies are in your `PYTHONPATH`.

**Comment by mdorier** [comment]
Isn't spack supposed to do that for me?

**Comment by adamjstewart** [comment]
If you run `spack load py-jupyterlab-server` it will, or if you `module load` the resulting module file.

**Comment by mdorier** [comment]
Ok I figured out the problem: I had actually created an environment, activated it, then added py-jupyterlab-server to it and executed `spack install`. I then saw the `jupyter` command available in the path so I tried running the notebook. As a matter of fact, I had to first deactive the environment, and reactivate it, to also have PYTHONPATH updated. For some reasons `spack install` inside an environment does not update PYTHONPATH. I'm closing the issue.
